chain/deals/provider.go

```diff
@@ -132,9 +135,12 @@ func NewProvider(ds dtypes.MetadataDS, sminer *storage.Miner, secb *sectorblocks
 	return h, nil
 }
 
-func (p *Provider) Run(ctx context.Context) {
+func (p *Provider) Run(ctx context.Context, host host.Host) {
 	// TODO: restore state
 
+	host.SetStreamHandler(storagemarket.DealProtocolID, p.HandleStream)
```

Comment:
see comment below -- I think host should be passed to the constructor and saved as a member. and then still put the SetStreamHandler calls in run.